gcp/table_gcp_dns_record_set.go

```diff
@@ -0,0 +1,209 @@
+				Description: "ForceSendFields is a list of field names (e.g. 'Kind') to unconditionally include in API requests. By default, fields with empty values are omitted from API requests. However, any non-pointer, non-interface field appearing in ForceSendFields will be sent to the server regardless of whether the field is empty or not. This may be used to include empty fields in Patch requests.",
+				Type:        proto.ColumnType_JSON,
+				Transform: transform.FromField("RecordSet.ForceSendFields"),
+			},
+			{
+				Name:        "null_fields",
+				Description: "NullFields is a list of field names (e.g. 'Kind') to include in API requests with the JSON null value. By default, fields with empty values are omitted from API requests. However, any field with an empty value appearing in NullFields will be sent to the server as null. It is an error if a field in this list has a non-empty value. This may be used to include null fields in Patch requests.",
+				Type:        proto.ColumnType_JSON,
+				Transform: transform.FromField("RecordSet.NullFields"),
+			},
+
+			// Steampipe standard columns
+			{
+				Name:        "title",
+				Description: ColumnDescriptionTitle,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromField("RecordSet.Name"),
+			},
+			{
+				Name:        "akas",
+				Description: ColumnDescriptionAkas,
+				Type:        proto.ColumnType_JSON,
+				Hydrate:     getDnsRecordSetAka,
+				Transform:   transform.FromValue(),
+			},
+
+			// Standard gcp columns
+			{
+				Name:        "location",
+				Description: ColumnDescriptionLocation,
+				Type:        proto.ColumnType_STRING,
+				Transform:   transform.FromConstant("global"),
+			},
+			{
+				Name:        "project",
+				Description: ColumnDescriptionProject,
+				Type:        proto.ColumnType_STRING,
+				Hydrate:     getProject,
+				Transform:   transform.FromValue(),
+			},
+		},
+	}
+}
+
+//// LIST FUNCTIONS
+
+func listDnsRecordSet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("listDnsRecordSet")
+
+	// Get the details of Cloud DNS Managed Zone
+	managedZone := h.Item.(*dns.ManagedZone)
+
+	// Create Service Connection
+	service, err := DnsService(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+
+	// Get Project details
+	projectData, err := activeProject(ctx, d)
+	if err != nil {
+		return nil, err
+	}
+	project := projectData.Project
+
+	resp, err := service.ResourceRecordSets.List(project, managedZone.Name).Do()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, recordset := range resp.Rrsets {
+		d.StreamListItem(ctx, recordSetInfo{recordset, managedZone.Name})
+	}
+
+	return nil, nil
+}
+
+//// HYDRATE FUNCTIONS
+
+func getDnsRecordSet(ctx context.Context, d *plugin.QueryData, h *plugin.HydrateData) (interface{}, error) {
+	plugin.Logger(ctx).Trace("getDnsRecordSet")
+
+	// Get the details of Cloud DNS Managed Zone
+	managedZone := h.Item.(*dns.ManagedZone)
```

Comment:
This is not required for `GET`, since we are expecting managed zone name as input parameter
```
KeyColumns: plugin.AllColumns([]string{"managedzone_name", "name", "type"}),
```